src/error.rs

```diff
@@ -0,0 +1,49 @@
+use goblin::error::Error as GoblinError;
+use pdb_crate::Error as PDBError;
+pub type CompactSymbolResult<T> = Result<T, CompactSymbolError>;
+
+#[derive(Fail, Debug)]
+pub enum CompactSymbolError {
+    #[fail(display = "Strings do not match")]
+    UnmatchedString,
```

Comment:
Let's be even more specific here and make this error describe the exact thing that mismatched, which is the breakpad ID. (I think we shouldn't try to find "categories" of errors - just give a separate enum value to every possible error occurrence, for maximum clarity.)
E.g. display = "Breakpad ID does not match", UnmatchedBreakpadID